src/test/java/com/sap/oss/phosphor/fosstars/model/score/oss/ArtifactReleaseHistoryScoreTest.java

```diff
@@ -239,12 +244,141 @@ public void testFourteenMonthOldSingleVersion() {
+    ArtifactVersion version110 = new ArtifactVersion("1.1.0", now.minusDays(6 * 31));
+    ArtifactVersion version120 = new ArtifactVersion("1.2.0", now.minusDays(5 * 31));
+    ArtifactVersion version130 = new ArtifactVersion("1.3.0-MIGHTY", now.minusDays(4 * 31));
+    ArtifactVersion version140 = new ArtifactVersion("1.4.0", now.minusDays(3 * 31));
+    ArtifactVersion version150 = new ArtifactVersion("1.5.0", now.minusDays(64));
+    ArtifactVersion version200 = new ArtifactVersion("2.0.0", now.minusDays(31));
+    ArtifactVersions artifactVersions = ArtifactVersions.of(version100, version110, version120,
+        version130, version140, version150, version200);
+
+    ArtifactReleaseHistoryScore score = new ArtifactReleaseHistoryScore();
+    Value<ArtifactVersions> versions = RELEASED_ARTIFACT_VERSIONS.value(artifactVersions);
+    Value<ArtifactVersion> version = ARTIFACT_VERSION.value(version120);
+    ScoreValue value = score.calculate(versions, version);
+    assertEquals(8.8, value.get(), DELTA);
+
+    final Optional<Value<?>> versionsValue =
+        value.usedValues().stream().filter(v -> v instanceof ArtifactVersionsValue).findFirst();
+    assertTrue(versionsValue.isPresent());
+    assertTrue(((ArtifactVersions) versionsValue.get().get()).containsAllOf(artifactVersions));
+  }
+
+  @Test
+  public void testVersionsSelectionFallBackIfOnlyOneMajorExists() {
+    LocalDateTime now = LocalDateTime.now();
+    ArtifactVersion version100 = new ArtifactVersion("1.0.0", now.minusDays(7 * 31));
+    ArtifactVersion version110 = new ArtifactVersion("1.1.0", now.minusDays(6 * 31));
+    ArtifactVersion version120 = new ArtifactVersion("1.2.0", now.minusDays(5 * 31));
+    ArtifactVersion version130 = new ArtifactVersion("1.3.0", now.minusDays(4 * 31));
+    ArtifactVersion version140 = new ArtifactVersion("1.4.0", now.minusDays(3 * 31));
+    ArtifactVersion version150 = new ArtifactVersion("1.5.0", now.minusDays(64));
+    ArtifactVersion version200 = new ArtifactVersion("2.0.0", now.minusDays(31));
+    ArtifactVersions artifactVersions = ArtifactVersions.of(version100, version110, version120,
+        version130, version140, version150, version200);
+
+    ArtifactReleaseHistoryScore score = new ArtifactReleaseHistoryScore();
+    Value<ArtifactVersions> versions = RELEASED_ARTIFACT_VERSIONS.value(artifactVersions);
+    Value<ArtifactVersion> version = ARTIFACT_VERSION.value(version200);
+    ScoreValue value = score.calculate(versions, version);
+    assertEquals(6.1, value.get(), DELTA);
+
+    final Optional<Value<?>> versionsValue =
+        value.usedValues().stream().filter(v -> v instanceof ArtifactVersionsValue).findFirst();
+    assertTrue(versionsValue.isPresent());
+    assertTrue(((ArtifactVersions) versionsValue.get().get()).containsAllOf(artifactVersions));
+  }
+
+  @Test
+  public void testVersionsSelectionIfNonSemanticVersionExists() {
```

Comment:
Isn't this test case a duplicate of `testVersionsSelectionByComparingMajor`?` 1.3.0-MIGHTY` is not a semantic version.

Reply:
`1.3.0-MIGHTY` is a semantic version right? I based this input from here https://github.com/SAP/fosstars-rating-core/blob/master/src/test/java/com/sap/oss/phosphor/fosstars/model/value/SemanticVersionTest.java#L27

Reply:
I agree that we need to replace `SemanticVersion`. We can do so with this with the java lib https://search.maven.org/artifact/com.vdurmont/semver4j/3.1.0/jar.

However I observed that this would completely restrict us from using versions like
- `1.2` which is a valid semantic version

And we have to be limited to versions like as given below. More information about the format [here](https://semver.org/#backusnaur-form-grammar-for-valid-semver-versions)
- `1.2.0`
- `1.2.0-SNAPSHOT` 

Then we also need to consider the possibility of using non-semantic versions. I believe we need to take this task of updating `SemanticVersion` there. I have created an issue for that #565